### PR TITLE
CS-42935/feat-taxonomy-and-eah

### PR DESCRIPTION
**Pull request by ishaileshmishra**
-  Taxonomy query support added
-  Early Access Feature Support

### CODE REVIEW

File: src/main/java/com/contentstack/sdk/Constants.java [hunk 1]
```diff
@@ -26,6 +26,8 @@ public class Constants {
     protected static final String CONTENT_TYPE_UID = "content_type_uid";
     protected static final String ENTRY_UID = "entry_uid";
     protected static final String LIVE_PREVIEW = "live_preview";
+
+    protected static final String TaxonomyPath = "/taxonomies/entries?query";
```
Comment (nadeem-cs) [inline src/main/java/com/contentstack/sdk/Constants.java:30]: Change this to all UPPERCASE with an underscore separating the words as per the naming conventions for constants. @ishaileshmishra

File: src/test/java/com/contentstack/sdk/TestAsset.java [hunk 2]
```diff
@@ -14,16 +14,14 @@ class TestAsset {
     private String assetUid;
     private final Stack stack = Credentials.getStack();
 
-    private String envChecker() {
+    private void envChecker() {
         String githubActions = System.getenv("GITHUB_ACTIONS");
         if (githubActions != null && githubActions.equals("true")) {
             System.out.println("Tests are running in GitHub Actions environment.");
             String mySecretKey = System.getenv("API_KEY");
             System.out.println("My Secret Key: " + mySecretKey);
```
Comment (nadeem-cs) [inline src/test/java/com/contentstack/sdk/TestAsset.java:22]: Why are we logging the API_KEY here? @ishaileshmishra

File: src/test/java/com/contentstack/sdk/Credentials.java [hunk 3]
```diff
@@ -65,7 +64,7 @@ private Credentials() throws AccessException {
     public static Stack getStack() {
         if (stack == null) {
             var envCheck = envChecker();
-            System.out.println(envCheck);
+            System.out.println("idendified env file detected " + envCheck);
```
Comment (nadeem-cs) [inline src/test/java/com/contentstack/sdk/Credentials.java:67]: Why are we logging the env here? @ishaileshmishra

**Review by nadeem-cs — CHANGES_REQUESTED** [review]

**Comment by ishaileshmishra** [comment]
Thanks for identifying the issues @nadeem-cs , I have made required changes. Please review again

**Review by nadeem-cs — APPROVED** [review]
Approved.

**Pull request merged**